kafka/consumer/group.py

```diff
@@ -1070,8 +1070,7 @@ def _message_generator(self):
                 self._update_fetch_positions(partitions)
 
             poll_ms = 1000 * (self._consumer_timeout - time.time())
-            if not self._fetcher.in_flight_fetches():
-                poll_ms = min(poll_ms, self.config['reconnect_backoff_ms'])
+            poll_ms = min(poll_ms, self.config['retry_backoff_ms'])
```

Comment:
I would probably do this as a one-liner combined with the line above (https://github.com/dpkp/kafka-python/pull/1824/files#diff-9da855eb60eed8ecc72ce217c25e576cR1072)... I think it's easier to mentally think through the calculated value that way

Reply:
I pushed this as an additional commit... feel free to either squash or if you really don't like it then just drop that particular commit.